contracts/src/instances/corn/CornBase.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.22;
+
+import { ERC20 } from "openzeppelin/token/ERC20/ERC20.sol";
+import { SafeERC20 } from "openzeppelin/token/ERC20/utils/SafeERC20.sol";
+import { ICornSilo } from "../../interfaces/ICornSilo.sol";
+import { IHyperdrive } from "../../interfaces/IHyperdrive.sol";
+import { HyperdriveBase } from "../../internal/HyperdriveBase.sol";
+import { CornConversions } from "./CornConversions.sol";
+
+// FIXME: Update this comment.
+//
+// FIXME: Make sure that the vault shares token is 0.
+//
+/// @author DELV
+/// @title CornBase
+/// @notice The base contract for the Corn Hyperdrive implementation.
+/// @dev This Hyperdrive implementation is designed to work with standard
+///      Corn vaults. Non-standard implementations may not work correctly
+///      and should be carefully checked.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+abstract contract CornBase is HyperdriveBase {
+    using SafeERC20 for ERC20;
+
+    /// @dev The Corn Silo contract. This is where the base token will be
+    ///      deposited.
+    ICornSilo internal immutable _cornSilo;
+
+    /// @notice Instantiates the CornHyperdrive base contract.
+    /// @param __cornSilo The Corn Silo contract.
+    constructor(ICornSilo __cornSilo) {
+        _cornSilo = __cornSilo;
+    }
+
+    /// Yield Source ///
+
+    /// @dev Accepts a deposit from the user in base.
+    /// @param _baseAmount The base amount to deposit.
+    /// @return The shares that were minted in the deposit.
+    /// @return The amount of ETH to refund. Since this yield source isn't
+    ///         payable, this is always zero.
+    function _depositWithBase(
+        uint256 _baseAmount,
+        bytes calldata // unused
+    ) internal override returns (uint256, uint256) {
```

Comment:
i imagine this specific implementation is only for the lbtc silo, right? there is another silo that accepts dai, sdai, usdc so we'd need to account for the multiple base types for those silos, right?

Reply:
It should work for all of them. It’s made to be generic enough to handle all of the assets supported on the Corn Silo. For some assets, we’d need a zap, but I don’t think anything would need to change to support other assets.